core/services/ardupilot_manager/firmware_download/FirmwareDownload.py

```diff
@@ -34,11 +36,25 @@ class Platform(str, Enum):
 
     Pixhawk1 = "Pixhawk1"
     Navigator = "Navigator"
+    SITL_ARM = "SITL_arm_linux_gnueabihf"
+    SITL_X86 = "SITL_x86_64_linux_gnu"
```

Comment:
This is unnecessary, you can have it as:
`SITL = detect_sitl_string()`

This will avoid all logic related to both SITL versions.

Reply:
But they are different binaries.

Reply:
Yes, since we are not going to run X86 in ARM and ARM in x86, it does not make sense to have both, we can have the string for comparison done in runtime and remove the extra enum and all the logic behind both.

Reply:
Ok. Got it now.

Reply:
The problem behind the actual approach is that we need to have the control over the architecture inside our code, and not inside the enum for free, making the logic unnecessary inside run_with_sitl. can you provide the diff or patch for such change ?

Reply:
Exactly, and this is how it's done. The SITL binaries are platform specific, like the firmware binaries are. Not having them on the Enums seems like creating one more logic for a problem that is exactly what the others are.

Reply:
Pushed last commit as patch. Let me know what you think.

Reply:
The archtecture from last update seems better, that's true.